socorro/cron/jobs/ftpscraper.py

```diff
@@ -335,6 +345,16 @@ def download(self, url):
         assert response.status_code == 200, response.status_code
         return response.content
 
+    def skip_json_file(self, json_url):
+        basename = os.path.basename(json_url)
+        for rule in self.config.json_files_to_ignore:
+            # If the rule was, for example, `*.foo.bar` then,
+            # the regex becomes `.\.foo\.bar`.
+            regex_string = re.escape(rule).replace('\\*', '.')
```

Comment:
I was lazy about the `^` and `$`. Didn't think it's particularly plausible to have a file called `something.mozinfo.json.other.json` (they have to be something `.json` [anyway to get near this](https://github.com/peterbe/socorro/blob/bug-1261164-skip-bogus-json-files-in-ftpscraper/socorro/cron/jobs/ftpscraper.py#L175).

But this new fancy `fnmatch` looks perfect! I'll just use that instead.